### ISSUE TITLE
Properly remove DataTemplates

### ISSUE THREAD

**Issue by ogasser** [tumi8/vermont#110]
We should go through the code to remove all DataTemplates code. PR #69 was a good first step, but there is more to remove :)

List of files with DataTemplates code:

- [ ] src/modules/ipfix/aggregator/BaseHashtable.cpp
- [ ] src/modules/ipfix/aggregator/BaseHashtable.h
- [ ] src/modules/ipfix/database/IpfixDbWriterMongo.cpp
- [ ] src/modules/ipfix/database/IpfixFlowInspectorExporter.cpp

Comments in files mentioning DataTemplates:

- [ ] src/common/ipfixlolib/ipfixlolib.c
- [ ] src/common/ipfixlolib/ipfixlolib.h
- [ ] src/modules/ipfix/TemplateBuffer.cpp
- [ ] src/modules/ipfix/IpfixPrinter.cpp

**Comment by nickbroon** [comment]
i might be mis-understanding the code, but in the case of src/modules/ipfix/aggregator/BaseHashtable.cpp and src/modules/ipfix/aggregator/BaseHashtable.h I think the dataTemplate references are actually just the regular templates that describe the data records in the hash tables. I don't believe they are actually represent IpfixDataTemplate. Perhaps it should be called Template in this use?

**Comment by ogasser** [comment]
@nickbroon yep, if we mean regular templates we should rename the DataTemplate to Template.

**Comment by nickbroon** [comment]
'dataTemplateInfo' could be replaced with 'templateInfo' using sed.
'dataTemplate' could _not_ be replace with 'template' as 'template' is C++ reserved word. It could be replace with 'Template' but that does follow camalCase convention for variables and would probably still be rather confusing. Suggestions welcome. (I think I'm inclined to think that dataTemplate is reasonable here as it's a template that describes ipfix data, which could be contrasted with say an optionTemplate which describes ipfix options. https://www.google.com/search?q=ipfix+%22data+template%22 )

**Comment by ogasser** [comment]
`dataTemplate` sounds good to me :+1:
